projects/ADuCM3029_demo_cn0428_cn0429/src/ADuCM3029_demo_cn0428_cn0429.cpp

```diff
@@ -0,0 +1,1307 @@
+*
+*Copyright 2015-2018(c) Analog Devices, Inc.
+*
+*All rights reserved.
+*
+*Redistribution and use in source and binary forms, with or without modification,
+*are permitted provided that the following conditions are met:
+*    - Redistributions of source code must retain the above copyright
+*      notice, this list of conditions and the following disclaimer.
+*    - Redistributions in binary form must reproduce the above copyright
+*      notice, this list of conditions and the following disclaimer in
+*      the documentation and/or other materials provided with the
+*      distribution.
+*    - Neither the name of Analog Devices, Inc. nor the names of its
+*      contributors may be used to endorse or promote products derived
+*      from this software without specific prior written permission.
+*    - The use of this software may or may not infringe the patent rights
+*      of one or more patent holders.  This license does not release you
+*      from the requirement that you obtain separate licenses from these
+*      patent holders to use this software.
+*    - Use of the software either in source or binary form, must be run
+*      on or directly connected to an Analog Devices Inc. component.
+*
+*THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+*PARTICULAR PURPOSE ARE DISCLAIMED.
+*
+*IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+*RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+*BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+*THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************
+**/
+
+#include <sys/platform.h>
+#include "adi_initialize.h"
+#include "ADuCM3029_demo_cn0428_cn0429.h"
+#include "adi_m355_gas_sensor.h"
+#include <stdio.h>
+#include <string.h>
+extern "C" {
+#include "RingBuffer.h"
+#include "cli.h"
+}
+
+using namespace adi_sensor_swpack;
+
+/* ================================================== */
+/* 	SYSTEM VARIABLES - handles, memory, error flags	  */
+/* ================================================== */
+ADI_UART_RESULT		  eUartResult;
+ADI_GPIO_RESULT 	  eGpioResult;
+ADI_PWR_RESULT        ePwrResult;
+SENSOR_RESULT   	  eSensorResult;
+
+/* Memory for the GPIO driver */
+uint8_t gpioMemory[ADI_GPIO_MEMORY_SIZE] = {0};
+
+/* Handle for the UART device. */
+ADI_UART_HANDLE hDevice;
+
+/* Memory for the UART driver. */
+static uint8_t UartDeviceMem[ADI_UART_MEMORY_SIZE];
+
+/* ================================================== */
+/* 						VARIABLES					  */
```

Comment:
If you use this multi-line comments in this manner, please try to fit all of them to have the length:
`
/* ========================================================================== */
/* ====================================VARIABLES============================= */
/* ========================================================================== */
`